batch_conversion.py

```diff
@@ -0,0 +1,254 @@
+        print('---------------------------------------')
+        for i in range(len(self.urdf_files)):
+            # clears any previous Geometry and Material references, so there is no conflict
+            importer.urdf2webots.parserURDF.Geometry.reference = {}
+            importer.urdf2webots.parserURDF.Material.namedMaterial = {}
+            configFile = os.path.splitext(self.urdf_files[i])[0] + '.json'
+            print('---------------------------------------')
+            print('Converting: ', self.urdf_files[i])
+            print('---------------------------------------')
+            try:
+                with open(configFile) as json_file:
+                    config = json.load(json_file)
+                importer.convert2urdf(
+                    inFile=self.urdf_files[i],
+                    outFile=self.protosPaths[i] if not config['robotName'] else self.protosPaths[i] + config['robotName'] + '.proto',
+                    normal=config['normal'],
+                    boxCollision=config['boxCollision'],
+                    disableMeshOptimization=OverrideCfg['disableMeshOptimization'] if Override else config['disableMeshOptimization'],
+                    enableMultiFile=config['enableMultiFile'],
+                    staticBase=config['staticBase'],
+                    toolSlot=config['toolSlot'],
+                    initRotation=config['initRotation'])
+                self.EndReportMessage['converted'].append(self.urdf_files[i])
+            except Exception as e:
+                print(e)
+                self.EndReportMessage['failed'].append(self.urdf_files[i])
+            print('---------------------------------------')
+
+    def print_end_report(self):
+        # print the Report
+        print('---------------------------------------')
+        print('------------End Report-----------------')
+        print('---------------------------------------')
+        print('Configs updated:')
+        print('---------------------------------------')
+        for config in self.EndReportMessage['updateConfigs']:
+            print(config)
+        print('---------------------------------------')
+        print('Configs created:')
+        print('---------------------------------------')
+        for config in self.EndReportMessage['newConfigs']:
+            print(config)
+        print('---------------------------------------')
+        print('Successfully converted URDF files:')
+        print('---------------------------------------')
+        for urdf in self.EndReportMessage['converted']:
+            print(urdf)
+        if len(self.EndReportMessage['failed']) > 0:
+            print('---------------------------------------')
+            print('Failedd URDF conversions:')
+            print('---------------------------------------')
+            for urdf in self.EndReportMessage['failed']:
+                print(urdf)
+
+    def create_test_world(self, spacing=3):
```

Comment:
A small description of what this function actually does would be welcome.

Reply:
will do